frontend/app/forms/motion-to-reduce-conviction/motion-to-reduce-conviction.web.component.js

```diff
@@ -1,5 +1,140 @@
 import React from "react";
+import FormThatPrints from "../inputs/form-that-prints.component";
+import Section from "../inputs/section.component.js";
+import TextInput from "../inputs/text-input.component";
+import Select from "../inputs/select.component.js";
+import Radio from "../inputs/radio.component.js";
+import GroupSelect from "../inputs/group-select.component.js";
+import {
+  petitionerRepresentativeOptions,
+  courtTypeOptions,
+  JusticeCourtList,
+  DistrictCourtList
+} from "../form-common-options/form-common-options";
 
-export default function MotionToReduceConviction_Web() {
-  return <div>Web component</div>;
+export default function MotionToReduceConviction_Web({ data }) {
+  const [currentConvictionDegree, setCurrentConvictionDegree] = React.useState(
+    ""
+  );
+  const [
+    reducedConvictionDegreeOptions,
+    setReducedConvictionDegreeOptions
+  ] = React.useState(defaultConvictionDegreeOptions);
+
+  React.useEffect(() => {
+    setCurrentConvictionDegree(data.case.convictionDegree);
+  }, [data.case.convictionDegree]);
+  React.useEffect(() => {
+    if (currentConvictionDegree) {
+      setReducedConvictionDegreeOptions(
+        reducedConvictionDegreeOptionsFromCurrentConviction(
+          currentConvictionDegree
```

Comment:
Here is how I am dynamically setting the conviction degree options (i.e. if petitioner has a _first degree_ felony, his/her options would only be _second_ and _third degree_):

1. petitioner's conviction is set in component state as `currentConvictionDegree`
2. `currentConvictionDegree` is passed to `reducedConvictionDegreeOptionsFromCurrentConviction` func in useEffect
3.  `reducedConvictionDegreeOptionsFromCurrentConviction` returns correct conviction degree options based on the petitioner's conviction degree (e.g. if petitioner conviction degree is `'First Degree'`, valid options would be `'Second Degree'` and `'Third Degree'`.

 I am terrible at naming things. 🤦🏻‍♂️ I know it's verbose, but wanted it to be descriptive. Let me know if there is a better name.

Reply:
I think it would be better not to hold this in state, but just to calculate it during render. It would simplify things a fair amount, including (if I'm reading this correctly) getting rid of both state values and both effects.

Reply:
Thanks. I was torn on whether to get component state involved here or not. I will refactor this later tonight.